backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/ConfirmationCodeModelMapper.java

```diff
@@ -17,11 +17,14 @@ public abstract class ConfirmationCodeModelMapper extends AbstractModelMapper {
 
 	@Nullable
 	public CollectionModel<ConfirmationCodeModel> toModel(String userId, @Nullable Iterable<ConfirmationCode> confirmationCodes) {
+		if (confirmationCodes == null) {
+			return CollectionModel.empty();
+		}
 		final var confirmationCodeModels = StreamSupport.stream(confirmationCodes.spliterator(), false)
-			.map(confirmationCode -> toModel(confirmationCode)).toList();
+			.map(this::toModel).toList();
 
 		return wrapCollection(CollectionModel.of(confirmationCodeModels), ConfirmationCodeModel.class);
-	}	
+	}
```

Comment:
I actually think this entire method should be removed, along with the controller method that calls it.

Exposing a method to allow the consumer to enumerate the confirmation codes for a user creates an exploitable loophole where a user's email can be validated without ever using OOBA.

Reply:
I created a workitem for this: https://dev.azure.com/DTS-STN/Canada%20Dental%20Care%20Plan/_workitems/edit/4235

Reply:
The controller method is exposed to add a `link` element to user's `confirmation-codes` resource:
https://github.com/DTS-STN/canadian-dental-care-plan/blob/c8478738b41d4b0330952b25e94ec3c483328e5a/backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/UserModelMapper.java#L48

I feel the controller method should return something like a 405 (Method Not Allowed) instead.

Reply:
Yes.. good point @nicholas-ly

Reply:
> I created a workitem for this: https://dev.azure.com/DTS-STN/Canada%20Dental%20Care%20Plan/_workitems/edit/4235

Okay, in that case I'll just close this PR and start on the next task.